packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -184,6 +185,17 @@ class ${class_name}:
         """
         parameter_values = [${measure_api_parameters}]
         request = self._create_measure_request(parameter_values)
-
-        for response in self._get_stub().Measure(request):
-            yield self._deserialize_response(response)
+        self._measure_response = self._get_stub().Measure(request)
+
+        try:
+            for response in self._measure_response:
+                yield self._deserialize_response(response)  
+        except grpc.RpcError as e:
+            if e.code() == grpc.StatusCode.CANCELLED:
+                print("Measure call has been cancelled.")
+            else:
+                raise      
+
+    def cancel(self) -> None:
+        if self._measure_response and self._measure_response.is_active():
+            self._measure_response.cancel()
```

Comment:
The measure response should be nullified upon cancellation and measurement completion.

Reply:
Sure. I have nullified the measure response in finally block.